### DISCUSSION TITLE
Different data on similar props

### DISCUSSION THREAD

**Discussion by paur94** [KevinVandy/material-react-table#451] [Q&A]
Hi there, 
First, I want to thank you for the work you do!
When I console data in prop onColumnOrderChange, I see my column headers in the array. But when I console data from onColumnPinningChange or onColumnVisibilityChange, I see functions. How can I get the same array with headings in these props?

**Comment by KevinVandy** [comment]
There seems to be a lot of confusion for how all of the on[State]Change props work. They work the same as any setState function in react. So they can have either a raw value for the new state, or a callback function with previous state.

This is explained a bit more in depth here:
https://www.material-react-table.com/docs/guides/table-state-management#add-side-effects-in-set-state-callbacks
